Implement ADD_BINDER_LIBRARY() CMake function

Oleh-Kravchenko opened the pull request:
ADD_BINDER_LIBRARY(NAME <name>
	HEADERS header1 [header2 ...])
	[CONFIG <config>]
	[BIND namespace1 [namespace2 ...]]
	[SKIP namespace1 [namespace2 ...]]
	[COMPILE_OPTIONS option1 [option2 ...]]
	[INCLUDES_DIRECTORIES directory1 [directory2 ...]]
	[LINK_LIBRARIES lib1 [lib2 ...]]
	[SOURCES source1 [source2 ...]]

Signed-off-by: Oleh Kravchenko <oleg@kaa.org.ua>

Oleh-Kravchenko committed:
Implement ADD_BINDER_LIBRARY() CMake function

ADD_BINDER_LIBRARY(NAME <name>
	HEADERS header1 [header2 ...])
	[CONFIG <config>]
	[BIND namespace1 [namespace2 ...]]
	[SKIP namespace1 [namespace2 ...]]
	[COMPILE_OPTIONS option1 [option2 ...]]
	[INCLUDES_DIRECTORIES directory1 [directory2 ...]]
	[LINK_LIBRARIES lib1 [lib2 ...]]
	[SOURCES source1 [source2 ...]]

Signed-off-by: Oleh Kravchenko <oleg@kaa.org.ua>

v-yatsenko commented on source/cmake/modules/BinderConfig.cmake.in:
```diff
@@ -1,12 +1,171 @@
+	RESULT_VARIABLE
+		RES
+	OUTPUT_STRIP_TRAILING_WHITESPACE)
+
```
Remove temporary file ${CMAKE_CURRENT_BINARY_DIR}/python_lib_dir.py

v-yatsenko reviewed (commented):
Please use clang code style for CMake files

Oleh-Kravchenko committed:
BinderConfig improvements

- don't create temporary file python_lib_dir.py
- store PYTHON_SITE_PACKAGES_DIR in CMake cache

Signed-off-by: Oleh Kravchenko <oleg@kaa.org.ua>

Oleh-Kravchenko committed:
Fix issue with CMAKE_CXX_FLAGS (it's can be empty)

Signed-off-by: Oleh Kravchenko <oleg@kaa.org.ua>

v-yatsenko committed:
Fixing crash on parsing std::function

Oleh-Kravchenko committed:
Depends on BINDER_SOURCES in ADD_BINDER_LIBRARY

Signed-off-by: Oleh Kravchenko <oleg@kaa.org.ua>

Oleh-Kravchenko committed:
Don't export Python binding

Exporting of Python binding doesn't make any sense, because it's do not
used for developing. Actually it is a plugin for Python. So remove it to
don't confuse CMake, when Python binding is not present.

Signed-off-by: Oleh Kravchenko <oleg@kaa.org.ua>
